archive/imo/imo1969_q1.lean

```diff
@@ -56,15 +55,30 @@ infinitely many of them.
 $a = 4*(2+b)^4$ should do. So $m = 2+b$.
 -/
 
-theorem imo1969_q1 : ∀ b : ℕ, ∃ a : ℕ, a ≥ b ∧ ∀ n : ℕ, ¬ prime (n^4 + a) :=
-assume b,
-have h1 : 1 < 2+b, by linarith,
-have b^2 ≥ b, by nlinarith,
-have h2 : 4*(2+b)^4 ≥ b, by nlinarith,
+/-- `good_nats` is an abbreviation for the subtype of natural numbers satisfying the
+condition in the problem statement, namely the `a : ℕ` such that `n^4 + a` is not prime
+for any `n : ℕ`. -/
+abbreviation good_nats : Type := {a : ℕ // ∀ n : ℕ, ¬ prime (n^4 + a)}
+
+/--
+The function $a = 4*(2+b)^4$ as a function from `ℕ` to `good_nats`.
+-/
+def a (b : ℕ) : good_nats :=
+⟨4*(2+b)^4, polynomial_not_prime (show 1 < 2+b, by linarith)⟩
```

Comment:
You should be able to prove that this is `strict_mono` (hence, `injective`) using dot-syntax lemmas. If not, this is a good reason to add more of them.

Reply:
Thanks! This was a pleasant exercise. I added `strict_mono.nat_pow` in `data.nat.basic`.